src/include/sof/audio/buffer.h

```diff
@@ -222,4 +235,11 @@ static inline void buffer_init(struct comp_buffer *buffer, uint32_t size)
 	buffer->avail = 0;
 	buffer_zero(buffer);
 }
+
+static inline void buffer_add_callback(struct comp_buffer *buffer,
+				       struct buffer_callback *cb)
+{
+	list_item_append(&buffer->callback_list, &cb->list);
```

Comment:
This should be the part of buffer_set_cb function with additional callback allocation logic.